library/core/src/any.rs

```diff
@@ -662,12 +664,39 @@ impl dyn Any + Send + Sync {
 /// While `TypeId` implements `Hash`, `PartialOrd`, and `Ord`, it is worth
 /// noting that the hashes and ordering will vary between Rust releases. Beware
 /// of relying on them inside of your code!
-#[derive(Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Debug, Hash)]
+#[derive(Clone, Copy, Debug, Hash, Eq)]
 #[stable(feature = "rust1", since = "1.0.0")]
 pub struct TypeId {
     t: u64,
 }
 
+#[stable(feature = "rust1", since = "1.0.0")]
+impl StructuralPartialEq for TypeId {}
```

Comment:
Per T-libs (see https://github.com/rust-lang/rust/issues/102371#issuecomment-1261052734) we're going to wait on `derive_const` (or similar) before accepting more changes like this to library.

@rustbot label +S-Blocked

Reply:
Please do not make `TypeId` "structural match" - instead, that *needs to be removed* - as per e.g. https://github.com/rust-lang/rust/issues/77125#issuecomment-1221451165 - sadly the one PR trying to do it was rejected and I'm not seeing a replacement just removing "structural match":
* #99189

Reply:
the impl `StructualPartialEq` was added because the test for [issue-73976](src/test/ui/consts/issue-73976-monomorphic.rs) failed as it used `matches!`.
It could be changed to a normal `==` though (I think).

Reply:
I've added a commit that does those changes.